contracts/commitments/batched_lpc_verifier.sol

```diff
@@ -196,18 +204,29 @@ library batched_lpc_verifier {
 
         z_offset = basic_marshalling.skip_length(skip_to_z(blob, offset));
         if( fri_params.step_list[0] != 1){
-            uint256[] memory eval;
+            uint256[4] memory eval4;
+            uint256[] memory V;
+            uint256[] memory U;
+            uint256   i;
 
             for (polynom_index = 0; polynom_index < fri_params.leaf_size;) {
-                eval = evaluation_points.length == 1? eval = evaluation_points[0]: eval = evaluation_points[polynom_index];
-                fri_params.batched_U[polynom_index] = polynomial.interpolate(
+                eval4 = evaluation_points.length == 1? evaluation_points[0]: evaluation_points[polynom_index];
+                U = polynomial.interpolate(
                     blob,
-                    eval,
+                    eval4_to_eval(eval4),
                     z_offset,
                     fri_params.modulus
                 );
                 z_offset = basic_marshalling.skip_vector_of_uint256_be(blob, z_offset);
 
+                fri_params.batched_U[polynom_index][0] = 
```

Comment:
Is it possible that eval4[0] < 1? Because this one will be re-set again in the next loop